query_executor.go

```diff
@@ -151,12 +156,15 @@ func (q *queryExecutor) do(ctx context.Context, qry ExecutableQuery, hostIter Ne
 			continue
 		}
 
-		iter = q.attemptQuery(ctx, qry, conn)
+		iter = q.attemptQuery(ctx, qry, it, conn)
+		// merge query metrics across multiple iterators
+		qm.merge(iter.metrics)
+		iter.metrics = qm
```

Comment:
Maybe we should add a function to merge an iterator state across retries and not put this here directly.